LiteCore/RevTrees/RevTree.cc

```diff
@@ -569,10 +569,7 @@ namespace litecore {
     }
 
     bool RevTree::hasNewRevisions() const {
-        for ( Rev* rev : _revs ) {
-            if ( rev->isNew() || rev->sequence == 0_seq ) return true;
-        }
-        return false;
+        return std::any_of(_revs.begin(), _revs.end(), [](Rev* rev) { return rev->isNew() || rev->sequence == 0_seq; });
```

Comment:
I'm interested in the clang analysis justification for this.  It doesn't seem any more readable than the former, so does it have some performance gains or something?

Reply:
LLVM's justification for preferring this is lacking. As far as I know it's about the same performance. To me, this makes the code more verbal on what exactly it is doing - the reader can see `any_of`, and a lambda function.